src/impy/kinematics.py

```diff
@@ -184,9 +189,47 @@ def _get_maximum_AZ(self):
         max_ind = a_val.index(max(a_val))
         return self.component_A[max_ind], self.component_Z[max_ind]
 
+    def _get_sample(self, nevt):
```

Comment:
This should be implemented with a multinomial distribution. https://en.wikipedia.org/wiki/Multinomial_distribution

You upfront how many events are to be generated and the fractions of elements in the composite target. You can then sample random numbers of events per combination from the multinomial distribution.
```py
rng = np.random.default_rng(model_seed)
array_of_nevents_per_fraction = rng.multinomial(nevt, fractions)
```
You should initialize the generator with the model seed, to make this reproducible.

Reply:
This sampling logic is probably also better placed in MCRun. This also solves the issue that CompositeTarget does not know how many events are to be generated, but MCRun does.

Reply:
`rng.multinomial` in MCRun now.